toolchain/semantics/semantics_node_kind.h

```diff
@@ -0,0 +1,64 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_SEMANTICS_SEMANTICS_NODE_KIND_H_
+#define CARBON_TOOLCHAIN_SEMANTICS_SEMANTICS_NODE_KIND_H_
+
+#include <cstdint>
+
+#include "common/ostream.h"
+
+namespace Carbon {
+
+class SemanticsNodeKind {
+ private:
+  // Note that this must be declared earlier in the class so that its type can
+  // be used, for example in the conversion operator.
+  enum class KindEnum : uint8_t {
+#define CARBON_SEMANTICS_NODE_KIND(Name, ...) Name,
+#include "toolchain/semantics/semantics_node_kind.def"
+  };
+
+ public:
+#define CARBON_SEMANTICS_NODE_KIND(Name, ...)       \
+  static constexpr auto Name()->SemanticsNodeKind { \
```

Comment:
Hm, looks like `clang-format` got a bit confused here. I guess this is not bad enough to be worth turning it off for this macro.

Reply:
Throwing comments in so it's not missed.

  // `clang-format` has a bug with spacing around `->` returns here. See
  // https://bugs.llvm.org/show_bug.cgi?id=48320 for details.